src/Examine.AzureDirectory/AzureReadOnlyLuceneDirectory.cs

```diff
@@ -0,0 +1,216 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using Azure.Storage.Blobs;
+using Examine.LuceneEngine;
+using Examine.LuceneEngine.Directories;
+using Lucene.Net.Store;
+using Directory = Lucene.Net.Store.Directory;
+
+namespace Examine.AzureDirectory
+{
+    public class AzureReadOnlyLuceneDirectory : AzureLuceneDirectory
+    {
+        private readonly string _cacheDirectoryPath;
+        private readonly string _cacheDirectoryName;
+        private string OldIndexFolderName;
+
+        public AzureReadOnlyLuceneDirectory(
+            string storageAccount,
+            string containerName,
+            string cacheDirectoryPath,
+            string cacheDirectoryName,
+            bool compressBlobs = false,
+            string rootFolder = null) : base(storageAccount, containerName, null, compressBlobs, rootFolder, true)
+        {
+            _cacheDirectoryPath = cacheDirectoryPath;
+            _cacheDirectoryName = cacheDirectoryName;
+            if (CacheDirectory == null)
+            {
+                Trace.WriteLine("INFO CacheDirectory null. Creating or rebuilding cache");
+                CreateOrReadCache();
+            }
+            else
+            {
+                CheckDirty();
+            }
+        }
+
+        protected override void GuardCacheDirectory(Lucene.Net.Store.Directory cacheDirectory)
+        {
+            //Do nothing
+        }
+        private void CreateOrReadCache()
+        {
+            var indexParentFolder = new DirectoryInfo(
+                Path.Combine(_cacheDirectoryPath,
+                    _cacheDirectoryName));
+            if (indexParentFolder.Exists)
+            {
+                var subDirectories = indexParentFolder.GetDirectories();
+                if (subDirectories.Any())
+                {
+                    var directory = subDirectories.FirstOrDefault();
+                    OldIndexFolderName = directory.Name;
+                    CacheDirectory = new SimpleFSDirectory(directory);
+                    _lockFactory = CacheDirectory.LockFactory;
+                }
+                else
+                {
+                    RebuildCache();
+                }
+            }
+            else
+            {
+                RebuildCache();
+
+            }
+        }
+
+        public void ResyncCache()
+        {
```

Comment:
Does any of this need to be locked? What happens if this is called by multiple threads? Seems there might be requirements to make more of the code in this PR threadsafe

Reply:
@Shazwazza I didnt hit any issues with that, as we always save cache in new location and dispose old one, don't think so it has to be thread safe. :)